encodings/alp/src/alp/array.rs

```diff
@@ -53,21 +48,23 @@ impl ALPArray {
             }
         }
 
-        let patches_dtype = patches.as_ref().map(|a| a.dtype().as_nullable());
         let mut children = Vec::with_capacity(2);
         children.push(encoded);
         if let Some(patch) = patches {
+            if !patch.dtype().eq_ignore_nullability(&dtype) || !patch.dtype().is_nullable() {
+                vortex_bail!(
+                    "ALP patches for array with dtype {} must be nullable, got {}",
+                    dtype,
+                    patch.dtype()
```

Comment:
```suggestion
                    "ALP patches dtype, {}, must be nullable version of array dtype, {}",
                    patch.dtype(),
                    dtype,
```

The message seems not quite right; maybe this is clearer?